gstd/src/lock/mutex.rs

```diff
@@ -217,6 +231,46 @@ unsafe impl<T> Sync for Mutex<T> {}
 /// ```
 pub struct MutexLockFuture<'a, T> {
     mutex: &'a Mutex<T>,
+    own_up_for: BlockCount,
+}
+
+impl<'a, T> MutexLockFuture<'a, T> {
+    /// Sets the maximum number of blocks the mutex lock can be owned by
+    /// some message before the ownership can be seized by another rival
+    pub fn own_up_for(self, block_count: BlockCount) -> Result<Self> {
+        if block_count == 0 {
+            Err(Error::ZeroMxLockDuration)
+        } else {
+            Ok(MutexLockFuture {
+                mutex: self.mutex,
+                own_up_for: block_count,
+            })
+        }
+    }
+
+    fn acquire_lock_ownership(
+        &mut self,
+        owner_msg_id: MessageId,
+        current_block: BlockNumber,
+    ) -> Poll<MutexGuard<'a, T>> {
+        let locked_by = unsafe { &mut *self.mutex.locked.get() };
```

Comment:
why we aren't satisfied with `get_mut()` fn call?

Reply:
and anywhere else

Reply:
We can't use `get_mut` as the `mutex` ref field in the `MutexLockFututre` structure is non-mutable. Making it so will break the logic around the `Mutex.lock` method which borrows mutex as non-mutable